src/core/CogniteJsonEditorOptions.ts

```diff
@@ -43,52 +45,44 @@ export class CogniteJsonEditorOptions implements JSONEditorOptions {
 
     public get templates(): any {
-        return getAllNodes()
-            .map(node => [node])// each node is put to its own array for reduce function
-            .reduce((acc: any[], singleNodeArr: any[]): any[] => {
-                const node = singleNodeArr.pop();
-
-                let templates = acc;
-
-                const transformToTemplate = (templateArr: any[], node: any) => {
-                    const key = extractField(node.key)
-
-                    const temp = {
-                        text: key,
-                        title: node.node.description,
-                        className: 'jsoneditor-type-object',
-                        field: key,
-                        value: node.data
-                    }
-                    templateArr.push(temp);
-
-                    /**
-                     * if node type is array or map
-                     * adding sample data as a template
-                     */
-                    if (node.node.type === "array" || node.node.type === "map") {
-                        const temp = {
-                            text: `${key}-sample`,
-                            title: `Add sample item to ${key}`,
-                            className: 'jsoneditor-type-object',
-                            field: `${key}-sample`,
-                            value: node.sample
-                        }
-                        templateArr.push(temp);
-                    }
-                }
-
-                if (acc.length === 1) { // transform first acc value
-                    const firstNode = acc.pop();
-                    templates = [];
-                    transformToTemplate(templates, firstNode);
-                }
+        const allTemplates: any = [];
+        getAllNodes().forEach(ele => {
+            const key = extractField(ele.key);
+            const template = {
+                text: key,
+                title: ele.node.description,
+                className: 'jsoneditor-type-object',
+                field: key,
+                value: ele.data
+            }
+            allTemplates.push(template);
+
+            if (ele.node.discriminator && ele.node.data) {
+                // If discriminator exists, add all sub types as templates
+                Object.entries(ele.node.data).forEach(([subKey, subVal]) => {
+                    const template = {
+                        text: `${key}-${subKey}`,
+                        title: `Add sample item to ${key}`,
+                        className: "jsoneditor-type-object",
+                        field: `${key}`,
+                        value: getJson(subVal as BaseNode),
+                    };
+                    allTemplates.push(template);
+                });
+            }
 
-                if (node) {
-                    transformToTemplate(templates, node);
+            if (ele.node.type === "array" || ele.node.type === "map") {
```

Comment:
can't we use DataType enum here? instead of hardcoded strings

Reply:
Fixed